src/algebra/monoid_algebra/degree.lean

```diff
@@ -0,0 +1,244 @@
+
+To see this in action, you can look at the triple
+`(sup_support_mul_le, max_degree_mul_le, le_min_degree_mul)`.
+-/
+
+variables {R A B ι : Type*}
+
+namespace add_monoid_algebra
+open_locale classical big_operators
+
+/-! ### Results about the `finset.sup` and `finset.inf` of `finsupp.support` -/
+
+section general_results_assuming_semilattice_sup
+
+section semiring
+
+variables [semiring R]
+
+lemma sup_support_add_le [semilattice_sup B] [order_bot B]
+  (D : A → B) (f g : add_monoid_algebra R A) :
+  (f + g).support.sup D ≤ (f.support.sup D) ⊔ (g.support.sup D) :=
+begin
+  refine (finset.sup_le (λ d ds, _)),
+  cases finset.mem_union.mp (finsupp.support_add ds) with df dg,
+  { exact (finset.le_sup df).trans le_sup_left },
+  { exact (finset.le_sup dg).trans le_sup_right }
+end
+
+lemma le_inf_degree_add [semilattice_inf B] [order_top B]
+  (D : A → B) (f g : add_monoid_algebra R A) :
+  f.support.inf D ⊓ g.support.inf D ≤ (f + g).support.inf D :=
+sup_support_add_le (λ a : A, order_dual.to_dual (D a)) f g
+
+section add_only
+
+variables [has_add A] [has_add B]
+
+lemma sup_support_mul_le [semilattice_sup B] [order_bot B]
+  [covariant_class B B (+) (≤)] [covariant_class B B (function.swap (+)) (≤)]
+  {D : A → B} (Dm : ∀ {a b}, D (a + b) ≤ D a + D b)
+  (f g : add_monoid_algebra R A) :
+  (f * g).support.sup D ≤ f.support.sup D + g.support.sup D :=
+begin
+  refine (finset.sup_le (λ d ds, _)),
+  obtain ⟨a, af, b, bg, rfl⟩ : ∃ a, a ∈ f.support ∧ ∃ b, b ∈ g.support ∧ d = a + b,
+  { simpa only [finset.mem_bUnion, finset.mem_singleton, exists_prop] using f.support_mul g ds },
+  refine Dm.trans (add_le_add _ _);
+  exact finset.le_sup ‹_›,
+end
+
+lemma le_inf_degree_mul [semilattice_inf B] [order_top B]
+  [covariant_class B B (+) (≤)] [covariant_class B B (function.swap (+)) (≤)]
+  {D : A → B} (Dm : ∀ {a b}, D a + D b ≤ D (a + b))
+  (f g : add_monoid_algebra R A) :
+  f.support.inf D + g.support.inf D ≤ (f * g).support.inf D :=
+order_dual.of_dual_le_of_dual.mpr $
+  sup_support_mul_le (λ a b, order_dual.of_dual_le_of_dual.mp Dm) f g
+
+end add_only
+
+lemma sup_support_list_prod_le [add_monoid A] [add_monoid B] [semilattice_sup B] [order_bot B]
+  [covariant_class B B (+) (≤)] [covariant_class B B (function.swap (+)) (≤)]
+  {D : A → B} (D0 : D 0 ≤ 0) (Dm : ∀ a b, D (a + b) ≤ D a + D b) :
+  ∀ l : list (add_monoid_algebra R A),
+    l.prod.support.sup D ≤ (l.map (λ f : add_monoid_algebra R A, f.support.sup D)).sum
+| [] := begin
+    rw [list.map_nil, finset.sup_le_iff, list.prod_nil, list.sum_nil],
+    exact λ a ha, by rwa [finset.mem_singleton.mp (finsupp.support_single_subset ha)]
+  end
+| (f::fs) := begin
+    rw [list.prod_cons, list.map_cons, list.sum_cons],
+    exact (sup_support_mul_le Dm _ _).trans (add_le_add_left (sup_support_list_prod_le _) _)
+  end
+
+lemma list_prod_le_sup_support_sum [add_monoid A] [add_monoid B] [semilattice_inf B] [order_top B]
```

Comment:
```suggestion
lemma le_inf_support_sum [add_monoid A] [add_monoid B] [semilattice_inf B] [order_top B]
```

Generally when you generate the `inf`  lemmas, you should just reorder the existing name components rather than adding new ones.